src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -1157,7 +1157,7 @@ private MethodArgumentInfo FindForEachPatternMethod(TypeSymbol patternType, stri
                 methodName,
                 arity: 0,
                 basesBeingResolved: null,
-                options: LookupOptions.Default,
+                options: LookupOptions.SearchInExtensionTypes,
```

Comment:
>SearchInExtensionTypes [](http://example.com/codeflow?start=39&length=22)

I am not sure how well this option is going to work here. It feels like for methods, we should look for extensions after overload resolution failed (as we do for regular extension methods). For example, let's say we found a non-extension match, but it has wrong signature. Are we going to look for extension members then?

Reply:
Clarified in another comment. In short, we want this lookup to include extension types (when the normal lookup is empty) and we'll also want to include methods from extension types when normal lookup yielded a method group but none of the methods were applicable. Will update the spec first.